android/src/main/java/app/notifee/core/NotificationManager.java

```diff
@@ -467,9 +468,14 @@
             task -> {
               if (notificationType == NOTIFICATION_TYPE_TRIGGER
                   || notificationType == NOTIFICATION_TYPE_ALL) {
-                NotifeeAlarmManager.cancelAllNotifications();
-                // delete all from database
-                WorkDataRepository.getInstance(getApplicationContext()).deleteAll();
+                task.continueWith(NotifeeAlarmManager.cancelAllNotifications()).addOnSuccessListener(t -> {
```

Comment:
What I am having trouble understanding here is why - when we are already supposed to be in a continuation and we *should* be executing successfully, we need to call cancelAllNotifications and then continue+onSuccess

I haven't looked (I need to) but is it because `NotifeeAlarmmanager.canacleAllNotifications()` is an async task so it hasn't finished otherwise, and the work must be chained?

Reply:
Replying to myself after more code-surfing:
https://github.com/invertase/notifee/blob/cc615819def97cba43ff4de29f15007b79bc5db5/android/src/main/java/app/notifee/core/NotifeeAlarmManager.java#L181

Yep, it's returning a Continuation, so I guess the chaining is vital here...